### ISSUE TITLE
Media object and text-overflow in v3.3.2

### ISSUE THREAD

**Issue by davinkevin** [twbs/bootstrap#15958]
I recently update my project from Bootstrap 3.2.x to 3.3.2 (last release) and I discover an important difference with the [Media Object](http://getbootstrap.com/components/#media) element.

I want to use in the media-heading the special less mixin .text-overflow() to add the following css attribute to the title : 

    .media-heading {
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    }

That works well (very well) in Bootstrap 3.2, you can see it in the following plnkr : [Media-List in 3.2](http://plnkr.co/edit/B7cGRbw8DxtnXgFhnmvM?p=preview)

But, since 3.3.x I can't use ellipsis and text-overflow inside header : [Media-List in 3.3.2](http://plnkr.co/edit/GAqClT8LZlSSSwduXEa0?p=preview)

You can compare the picture below : 
![View in Bootstrap 3.2.0][1]
![View in Boostrap 3.3.2][2]
[1]: http://i.stack.imgur.com/75nOx.png
[2]: http://i.stack.imgur.com/ww4sD.png

It seems to be a regression, linked to the solution of #14990 
This behavior is very frustrating, especially on mobile device.

I've opened a question on [stackoverflow](http://stackoverflow.com/questions/28660494/media-list-and-text-overflow-in-bootstrap-3-3-2) and the natural solution was to go back to the media-list css of version 3.2 (display : block and width : auto).

Thanks for your help.

**Comment by twbs-lmvtfy** [comment]
Hi @davinkevin!

You appear to have posted a live example (http://run.plnkr.co/plunks/B7cGRbw8DxtnXgFhnmvM/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W001](https://github.com/twbs/bootlint/wiki/W001): `<head>` is missing UTF-8 charset `<meta>` tag
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* [W005](https://github.com/twbs/bootlint/wiki/W005): Unable to locate jQuery, which is required for Bootstrap's JavaScript plugins to work
* line 15, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 26, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 37, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 5, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0
* line 6, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.2.0

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @davinkevin!

You appear to have posted a live example (http://run.plnkr.co/plunks/GAqClT8LZlSSSwduXEa0/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W001](https://github.com/twbs/bootlint/wiki/W001): `<head>` is missing UTF-8 charset `<meta>` tag
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* [W005](https://github.com/twbs/bootlint/wiki/W005): Unable to locate jQuery, which is required for Bootstrap's JavaScript plugins to work
* line 15, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 26, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 37, column 9: [W010](https://github.com/twbs/bootlint/wiki/W010): Using `.pull-left` or `.pull-right` as part of the media object component is deprecated as of Bootstrap v3.3.0. Use `.media-left` or `.media-right` instead.
* line 5, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.3.1
* line 6, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.2 ; saw what appears to be usage of Bootstrap 3.3.1

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by davinkevin** [comment]
W001, W002, W003, W005 Solved.
W010 Solved in 3.3.2 (not in 3.2 of course)
W013 Solved in 3.3.2 (not in 3.2 of course)

The ellipsis still don't appear.

Thanks

**Comment by kkirsche** [comment]
I'm pretty sure this is caused by [`display: table-cell`](https://github.com/twbs/bootstrap/blob/4d2f68c4376741d1b261fd172d64518662383a63/less/media.less#L37) on `.media-body` which was not there in 3.2.0

**Comment by stubbornella** [comment]
Overflow hidden crops off fly out menus and tooltips, whereas table-cell doesn't play nicely with ellipsis. 




A flex box solution would probably solve for both but the IE support is pretty bad.




This isn't a case where there is a perfect solution. They all have advantages and drawbacks.

**Comment by bslugger** [comment]
I agree with kkirsche, I believe that width: 10000px is breaking the word wrap. Here is a JSbin of the text-overflow not working correctly in media-body to further explain the extent of this bug:
http://jsbin.com/lofuwu/1/edit?html,output 

I'm on Chrome Version 41.0.2272.89 (64-bit) on OSX Yosemite 10.10.2.

**Comment by mdo** [comment]
Not sure what we could do to change that at this point—the media `width` is the only way to accomplish that particular layout as I recall.

**Comment by mdo** [comment]
This might end up being a won't fix in v3.

**Comment by mdo** [comment]
Going for the won't fix for now.

**Comment by mschipperheyn** [comment]
This bug also appears when nowrap is not used, and the text is not wrappable, such as with a long url
